src/util-cpu.c

```diff
@@ -75,9 +76,15 @@ uint16_t UtilCpuGetNumProcessorsConfigured(void)
 
     return (uint16_t)nprocs;
 #elif OS_WIN32
-	long nprocs = -1;
-	const char* envvar = getenv("NUMBER_OF_PROCESSORS");
-	nprocs = (NULL != envvar) ? atoi(envvar) : 0;
+    int64_t nprocs = 0;
+    const char* envvar = getenv("NUMBER_OF_PROCESSORS");
+    if (envvar != NULL) {
+        if (StringParseInt64(&nprocs, 10, 0, envvar) < 0) {
+            SCLogWarning(SC_ERR_INVALID_VALUE, "Invalid value for number of "
+                         "processors: %s, resetting to 0", envvar);
+            nprocs = 0;
```

Comment:
resseting to 0 will trigger the next check right away? `if (nprocs < 1) {`.

Reply:
Yes and it'll give an additional error message to make things clearer:
```
 89         SCLogError(SC_ERR_SYSCALL, "Couldn't retrieve the number of cpus "
 90                    "configured from the NUMBER_OF_PROCESSORS environment variable");
```
Should the same error message and return statement be in the `StringParse` check block?

Reply:
I don't like the sequence of "hey you have an error we're setting something to 0" directly followed by "hey 0 isn't valid, we're going to use 1". If 0 isn't valid we shouldn't be setting it.

So I think the Warning should say we're setting to 1 and then set to 1 :)

Reply:
> I don't like the sequence of "hey you have an error we're setting something to 0" directly followed by "hey 0 isn't valid, we're going to use 1". If 0 isn't valid we shouldn't be setting it.
> 
Yes, that does look weird.

> So I think the Warning should say we're setting to 1 and then set to 1 :)
Didn't get this part. 1 is a valid value for number of processors but we failed so why should we set to 1? Or was it a figure of speech?

Reply:
Ah I misread the original code. Forget the setting to 1 part. Still this check should just issue a single warning if the value is invalid.

Reply:
OK. Thanks!